ocaml/typing/typecore.ml

```diff
@@ -3511,8 +3511,9 @@ let check_recursive_class_bindings env ids exprs =
          raise(Error(expr.cl_loc, env, Illegal_class_expr)))
     exprs
 
+
 (* Is the return value annotated with "local_" *)
-let is_local_returning_expr e =
+let is_local_returning_expr, is_local_returning_case_rhs =
```

Comment:
I think we should write a test for this. Might be worth putting them in another file so the Jane Street-specific ones are more cleanly separable from the existing tests when we send this upstream.

Reply:
Added to a file called `jane_test.ml` in the pattern guards test directory